app/routes/jobs.$jobId.tsx

```diff
@@ -45,7 +85,11 @@ export default function JobDetails() {
   const [errorMessageModalOpen, setErrorMessageModalOpen] = useState(false);
   const [expandedErrors, setExpandedErrors] = useState<Set<number>>(new Set());
   const [expandedModalErrors, setExpandedModalErrors] = useState<Set<number>>(new Set());
+  const [externalVerifierStatuses, setExternalVerifierStatuses] = useState<ExternalVerifierStatusMap>({});
+  const externalVerifierStatusesRef = useRef<ExternalVerifierStatusMap>({});
   const { serverUrl } = useServerConfig();
+  const hasExternalVerificationData = hasAllRequiredExternalVerifications(jobData?.externalVerifications);
+  const isJobFullyCompleted = Boolean(jobData?.isJobCompleted) && hasExternalVerificationData;
```

Comment:
Is there a scenario where we wouldn't store data for all external verifiers? I think requiring the data to be present isn't perfect.

Reply:
> Is there a scenario where we wouldn't store data for all external verifiers? I think requiring the data to be present isn't perfect.

I think we don't have to support the case in which we change settings in Sourcify (e.g. we disable a specific verifier). UI should just mirror whatever configuration is currently in use. If we change the config of server, then we manually update also UI. The alternative would be to returned the list of enabled storage services so the UI can adapt.

> I found this also makes the page refetching data for old jobs which don't have external verifications for an indefinite amount of time.

Right it makes sense. I think we should limit with a maximum amount of re-fetch